frontend/public/components/debug-terminal.tsx

```diff
@@ -0,0 +1,226 @@
+  FirehoseResult,
+  LoadingBox,
+  HorizontalNav,
+  PageHeading,
+} from '@console/internal/components/utils';
+import { PodKind, k8sCreate, k8sKillByName } from '@console/internal/module/k8s';
+import { PodExecLoader } from '@console/internal/components/pod';
+import { PodModel } from '@console/internal/models';
+
+import { resourcePath } from './utils/resource-link';
+import { getBreadcrumbPath } from '@console/internal/components/utils/breadcrumbs';
+
+const getDebugPod = async (
+  debugName: string,
+  podToDebug: PodKind,
+  containerName: string,
+): Promise<PodKind> => {
+  const container = podToDebug?.spec?.containers?.find((c) => c.name === containerName);
+  return {
+    kind: 'Pod',
+    apiVersion: 'v1',
+    metadata: {
+      name: debugName,
+      namespace: podToDebug?.metadata?.namespace,
+      annotations: {
+        'debug.openshift.io/source-container': containerName,
+        'debug.openshift.io/source-resource': `/v1, Resource=pods/${podToDebug?.metadata?.name}`,
+      },
+    },
+    spec: {
+      activeDeadlineSeconds: 21600,
+      volumes: [],
+      containers: [
+        {
+          name: containerName,
+          image: container?.image,
+          command: ['/bin/sh'],
+          resources: {},
+          volumeMounts: [],
+          securityContext: {
+            privileged: true,
+            runAsUser: 0,
+          },
+          stdin: true,
+          stdinOnce: true,
+          tty: true,
+        },
+      ],
+      restartPolicy: 'Never',
+      name: podToDebug?.metadata?.name,
+      hostNetwork: true,
+      hostPID: true,
+    },
+  };
+};
+
+const DebugTerminalError: React.FC<DebugTerminalErrorProps> = ({ error }) => {
+  return (
+    <div className="co-m-pane__body">
+      <Alert variant="danger" isInline title={error} />
+    </div>
+  );
+};
+
+const DebugTerminalInner: React.FC<DebugTerminalInnerProps> = ({ obj, activeContainer }) => {
+  const { t } = useTranslation();
+  const infoMessage = (
+    <div className="padding-info-message">
+      <Alert
+        variant="info"
+        isInline
+        title={t(
+          'public~This temporary pod has a modified entrypoint command to debug a failing container. The pod will be available for one hour and will be deleted when the terminal window is closed.',
+        )}
+      />
+    </div>
+  );
+  switch (obj?.data?.status?.phase) {
+    case 'Failed':
+      return (
+        <DebugTerminalError
+          error={
+            <>
+              {t('public~The debug pod failed. ')}
+              {obj?.data?.status?.containerStatuses?.[0]?.state?.terminated?.message ||
+                obj?.data?.status?.message}
+            </>
+          }
+        />
+      );
+    case 'Running':
+      return (
+        <PodExecLoader obj={obj.data} activeContainer={activeContainer} infoMessage={infoMessage} />
+      );
+    default:
+      return <LoadingBox />;
+  }
+};
+
+export const DebugTerminal: React.FC<DebugTerminalProps> = (props) => {
```

Comment:
Destructure `props` to dry up a the code below a bit?
```suggestion
export const DebugTerminal: React.FC<DebugTerminalProps> = ({ obj, match: { params: { name } }) => {
```